### PR TITLE
DEVPROD-8550: Fix overnight sleep schedule calculation

### PR DESCRIPTION
**Pull request by sophstad**
DEVPROD-8550

### Description

Fix calculation of sleep schedule when host is set to run overnight

### Testing

Add unit tests for calculator function

### CODE REVIEW

File: apps/spruce/src/components/Spawn/utils/hostUptime.test.ts [hunk 1]
```diff
@@ -549,3 +550,89 @@ describe("getNextHostStart", () => {
     });
   });
 });
+
+describe("getEnabledHoursCount", () => {
```
Comment (khelif96) [inline apps/spruce/src/components/Spawn/utils/hostUptime.test.ts:554]: Are there any possible invalid inputs this function could get passed for example `stopTime` is before `startTime` that we should test against.

Comment (sophstad) [reply apps/spruce/src/components/Spawn/utils/hostUptime.test.ts:554]: `stopTime` before `startTime` is exactly what this PR implements to allow overnight schedules 😅

File: apps/spruce/src/components/Spawn/utils/hostUptime.ts [hunk 2]
```diff
@@ -161,7 +161,17 @@ const getDailyUptime = ({
 }: {
   startTime: string;
   stopTime: string;
-}) => differenceInHours(new Date(stopTime), new Date(startTime));
+}) => {
+  const startDate = new Date(startTime);
+  const stopDate = new Date(stopTime);
+
+  // If this is an overnight schedule, set stop date to the following day so that uptime is correctly calculated.
+  if (stopDate < startDate) {
```
Comment (SupaJoon) [inline apps/spruce/src/components/Spawn/utils/hostUptime.ts:169]: How does this condition indicate an overnight schedule?

Comment (sophstad) [reply apps/spruce/src/components/Spawn/utils/hostUptime.ts:169]: If the stopTime occurs before the startTime, the back end interprets the start as occurring on the following day

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**